pkg/ottl/ottlfuncs/func_remove_xml.go

```diff
@@ -82,7 +82,7 @@ func parseNodesXML(targetVal string) (*xmlquery.Node, error) {
 	if err != nil {
 		return nil, fmt.Errorf("parse xml: %w", err)
 	}
-	if !preserveDeclearation {
+	if !preserveDeclearation && top.FirstChild != nil {
```

Comment:
Is this change related?

Reply:
Not strictly, but it was noticed while adding tests for this. I'll move it over to #35463.